src/useErrorBoundary.ts

```diff
@@ -2,15 +2,25 @@ import { useContext, useMemo, useState } from "react";
 import { assertErrorBoundaryContext } from "./assertErrorBoundaryContext";
 import { ErrorBoundaryContext } from "./ErrorBoundaryContext";
 
+type StateError<TError> =
+  | (TError & { isShowBoundary: true })
+  | (Error & { isShowBoundary: true });
+
 type UseErrorBoundaryState<TError> =
-  | { error: TError; hasError: true }
+  | { error: StateError<TError>; hasError: true }
   | { error: null; hasError: false };
 
 export type UseErrorBoundaryApi<TError> = {
   resetBoundary: () => void;
   showBoundary: (error: TError) => void;
 };
 
+function normalize<TError>(error: TError): StateError<TError> {
+  return typeof error === "object"
+    ? Object.assign(Object.create(error as object), { isShowBoundary: true })
+    : Object.assign(Error(error as string), { isShowBoundary: true });
+}
```

Comment:
Using `Object.create` to work around non-extensible errors seems nice. I _think_ `instanceof` checks will still work as expected with this.

That being said, I think we might be able to avoid cloning error objects unnecessarily here if were to change this to check `Object.isExtensible`? I'd also change the name to something more descriptive:

```suggestion
function suppressLoggingForError(error: any): any {
  if (error != null) {
    switch (typeof error) {
      case "object": {
        if (Object.isExtensible(error)) {
          error._suppressLogging = true;

          return error;
        } else {
          return Object.assign(Object.create(error as object), {
            _suppressLogging: true
          });
        }
      }
      case "string": {
        return bject.assign(Error(error as string), { _suppressLogging: true });
      }
    }
  }

  return error;
}
```

Note _anything_ can be thrown in JavaScript so I think an `any` type is reasonable here. Also note that we could try to handle other types here but it's probably not worth it. In most cases, the value being thrown is probably either a string or an Error.

Reply:
Rename the function from `normalize`  to `suppressLoggingForError` and adopt the suggested logic.